File: python/msgpackrpc/client/twisted/future.py
class Future(object):
    """
    This class is used as the result of asynchronous call.
    By using join(), the caller is able to wait for the completion.
    """

    def __init__(self, loop):
        self._error = None
        self._result = None
        self._loop = loop

    def join(self):
        # TODO(kzk): implement client-side timeout
        while (self._error == None and self._result == None):
            self._loop.run()

    def get_result(self):
        return self._result

    def set_error(self, error):
        self._error = error

    def set_result(self, result):
        self._result = result

File: php/MessagePackRPC/Future.php
<?php
class MessagePackRPC_Future
{
  public $result = null;
  public $errors = null;
  public $loop   = null;

  public function __construct($loop)
  {
    $this->loop = $loop;
  }

  public function setResult($result)
  {
    $this->result = $result;
  }

  public function getResult()
  {
    $result = $this->result;
    return $result;
  }

  public function setErrors($errors)
  {
    $this->errors = $errors;
  }

  public function getErrors()
  {
    $errors = $this->errors;
    return $errors;
  }

  public function join()
  {
    // TODO: Event Loop Implementation

    // while (($this->errors == null) && ($this->result == null)) {
    //   $this->loop->run();
    // }
  }
}